entries/hgrosser/src/1brc.pas

```diff
@@ -95,36 +96,73 @@ function outTimer(start: int64): NumStr;
 end; {outTimer}
 
 procedure error_halt(s: ansistring);
-  {outputs a Fatal Error and halt's the program}
+{outputs a Fatal Error and halt's the program}
 begin
   writeln('FATAL ERROR!');
   writeln(s);
   halt(1);
 end;
 
+{$ASMMODE INTEL}
```

Comment:
My guess is that a proper pure pascal function, once inlined, may be faster than this asm, which is for Linux x86_64 only.

Or at least use `buf` instead of `RDI`, and `len` instead of `ecx` within the loop: the FPC asm would work also on Win64 (and Win32).

Reply:
What is the issue with the 'FATAL ERROR!'? It is never called, if Conditional XTEST is not defined, as it should be.

The assembler hash is on my computer 3 seconds faster than a pure pascal function. I will change RDI and ecx as you requested.

Reply:
It was not about this source code line: I don't know what github did with my comment.
It was about the hash.

Reply:
Hey Hartmut(@hg747),

As long as a PR is open you can commit to it as many commits you think are needed to make it ok.

One PR is not bound by a single commit. GitHub will group all the commit you push to it and they will all be merged once we agree that things are good.

Am I making sense?

Cheers,
Gus

Reply:
Hey Hartmut(@hg747),

Are we good to merge?

Cheers,
Gus

Reply:
Thank you Gus for clarification about PR's and multiple commits. From my side we are ready to merge.